repository/src/test/java/by/zhigarev/dao/impl/TagDAOImplTest.java

```diff
@@ -0,0 +1,104 @@
+package by.zhigarev.dao.impl;
+
+
+import by.zhigarev.dao.TagDAO;
+import by.zhigarev.dao.exception.DAOException;
+import by.zhigarev.dao.util.TagMapper;
+import by.zhigarev.model.Tag;
+import org.junit.jupiter.api.*;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabase;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TagDAOImplTest {
+    private EmbeddedDatabase database;
+    private TagDAO tagDAO;
+    private TagMapper tagMapper = new TagMapper();
+
+    private static final String TEST_NAME = "Test Name";
+    private static final String DB_NAME_FIRST = "First Tag";
+    private static final String DB_NAME_SECOND = "Second Tag";
+    private static final String DB_NAME_THIRD = "Third Tag";
+    private static final String TEST_NEW_NAME = "New Name";
+    private static final int TEST_ID_FIRST = 1;
+    private static final int TEST_ID_SECOND = 2;
+    private static final int TEST_ID_THIRD = 3;
+
+    @BeforeEach
+    void setUp() {
+        database = new EmbeddedDatabaseBuilder()
```

Comment:
Instead of creating database in setup method please use separate test config for this purpose, the same for DAO.